### PR TITLE
feat: teiParams are ACL checked DHIS2-14604]

### PR DESCRIPTION
**Pull request by gnespolino**
**DON'T MERGE UNTILL TESTS ARE GREEN**

### CODE REVIEW

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/tei/TeiQuerySecurityManager.java [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+import org.hisp.dhis.feedback.ErrorCode;
+import org.hisp.dhis.organisationunit.OrganisationUnit;
+import org.hisp.dhis.user.CurrentUserService;
+import org.hisp.dhis.user.User;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class TeiQuerySecurityManager
```
Comment (maikelarabori) [inline dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/tei/TeiQuerySecurityManager.java:67]: Would be nice to have a Javadoc for this class.

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/tei/TeiQuerySecurityManager.java [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+{
```
Comment (maikelarabori) [inline dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/tei/TeiQuerySecurityManager.java:68]: It looks like all methods in this class declare TeiQueryParams, but they only use CommonQueryParams. All methods could receive just CommonQueryParams. They can be extracted and reused later when we start refactoring events/enrollments queries.

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/tei/TeiQuerySecurityManager.java [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+            // -----------------------------------------------------------------
+            dimension.getItems().clear();
+            dimension.getItems().addAll( canReadItems );
+
+            log.debug( String.format( "User: '%s' constrained by dimension: '%s'", user.getUsername(),
+                dimension.getDimension() ) );
+        }
+    }
+
+    private boolean hasDimensionOrFilterWithItems( TeiQueryParams teiQueryParams, String dimensionUid )
```
Comment (maikelarabori) [inline dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/tei/TeiQuerySecurityManager.java:277]: Would be nice a Javadoc on this method.

File: dhis-2/dhis-services/dhis-service-schema/src/main/java/org/hisp/dhis/schema/descriptors/TrackedEntityAttributeSchemaDescriptor.java [hunk 4]
```diff
@@ -52,6 +52,7 @@ public Schema getSchema()
         Schema schema = new Schema( TrackedEntityAttribute.class, SINGULAR, PLURAL );
         schema.setRelativeApiEndpoint( API_ENDPOINT );
         schema.setOrder( 1450 );
+        schema.setDataShareable( true );
```
Comment (maikelarabori) [inline dhis-2/dhis-services/dhis-service-schema/src/main/java/org/hisp/dhis/schema/descriptors/TrackedEntityAttributeSchemaDescriptor.java:55]: I'm not sure about the consequences of this change...

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/tei/TeiQuerySecurityManager.java (hunk 1 above)
Comment (maikelarabori) [inline dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/tei/TeiQuerySecurityManager.java:67]: Would be important to write down some tests for this class.

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/security/DefaultAnalyticsSecurityManager.java [hunk 5]
```diff
@@ -104,6 +104,15 @@ public void decideAccess( DataQueryParams params )
         decideAccessDataReadObjects( params, user );
     }
 
+    @Override
+    public void decideAccess( List<OrganisationUnit> queryOrgUnits, Set<IdentifiableObject> readObjects )
```
Comment (maikelarabori) [inline dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/security/DefaultAnalyticsSecurityManager.java:108]: Would be cool to write some tests for this method.

**Review by maikelarabori — COMMENTED** [review]
Looks good @gnespolino :+1: . I left some comments/suggestions for possible improvements.

Cheers

**Review by d-bernat — APPROVED** [review]

**Pull request merged**